AbletonMCP_Remote_Script/handlers/midi.py

```
@@ -0,0 +1,411 @@
+"""MIDI: get notes (legacy + extended), quantize, transpose, clear, capture, groove."""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import traceback
+
+
+def get_clip_notes(song, track_index, clip_index, start_time, time_span, start_pitch, pitch_span, ctrl=None):
+    """Get MIDI notes from a clip."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)